backend/common/src/main/java/ai/verta/modeldb/common/db/migration/Migration.java

```diff
@@ -3,25 +3,35 @@
 import lombok.Value;
 
 @Value
-public class Migration implements Comparable<Migration> {
+class Migration implements Comparable<Migration> {
   String filename;
 
   @Override
   public int compareTo(Migration o) {
-    return this.getNumber() - o.getNumber();
+    int versionDiff = this.getNumber() - o.getNumber();
+    if (versionDiff == 0) {
+      if (this.isUp() && o.isUp()) {
+        return 0;
+      }
+      if (this.isDown()) {
+        return 1;
+      }
+      return -1;
+    }
+    return versionDiff;
   }
 
-  public int getNumber() {
+  int getNumber() {
     String[] pieces = filename.split("_");
     return Integer.parseInt(pieces[0]);
   }
 
-  public boolean isUp() {
+  boolean isUp() {
```

Comment:
I'm not clear on what `up` and `down` mean here. I can see that it is part of the filename, but I don't follow the meaning. Is that documented elsewhere?

Reply:
In the design doc: https://github.com/VertaAI/docs/blob/main/design-docs/2022-09-migrations.md

Reply:
Got it, that's kind of what I was expecting. At first I had it in my head that we were trying to avoid any rollbacks, which is why I wasn't sure. But then I realized it was that we don't want to have any **changes** to existing migrations, not that we don't provide rollback options.

Reply:
Maybe add javadoc explanation? Folks with access do this repo (open source) don't all have access to the `docs` repo.

Reply:
good idea. will add!

Reply:
done!